src/relay/ir/debug_printer.cc

```diff
@@ -0,0 +1,305 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file debug_printer.cc
+ * \brief A pretty printer for the Relay IR.
+ * As we had not determined a formal syntax yet, right now it is only for debug purpose.
+ */
+
+#include <tvm/relay/expr_functor.h>
+#include <tvm/relay/environment.h>
+#include <tvm/relay/error.h>
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <iostream>
+#include "../pass/type_functor.h"
+#include "doc.h"
+
+namespace tvm {
+namespace relay {
+
+using namespace tvm::runtime;
+
+struct PrintError : public Error {
+  explicit PrintError(const std::string& msg) : Error("PrintError: " + msg) {}
+};
+
+Doc KindDocify(TypeParamNode::Kind k) {
+  switch (k) {
+    case TypeParamNode::kShapeVar:
+      return DocOfStr("ShapeVar");
+    case TypeParamNode::kShape:
+      return DocOfStr("Shape");
+    case TypeParamNode::kBaseType:
+      return DocOfStr("BaseType");
+    case TypeParamNode::kType:
+      return DocOfStr("Type");
+    default:
+      throw PrintError("unreachable code");
```

Comment:
just do LOG(FATAL), as it throws and also gives you stack trace and LOC in the file for debugging purpose